src/fastfield/serializer/mod.rs

```diff
@@ -0,0 +1,164 @@
+use crate::common::BinarySerializable;
+use crate::common::CompositeWrite;
+use crate::common::CountingWriter;
+use crate::directory::WritePtr;
+use crate::schema::Field;
+use fastfield_codecs::CodecId;
+//pub use bitpacked::BitpackedFastFieldSerializer;
+pub use fastfield_codecs::bitpacked::BitpackedFastFieldSerializer;
+use fastfield_codecs::linearinterpol::LinearInterpolFastFieldSerializer;
+pub use fastfield_codecs::FastFieldDataAccess;
+pub use fastfield_codecs::FastFieldSerializerEstimate;
+pub use fastfield_codecs::FastFieldStats;
+use std::io::{self, Write};
+
+/// `CompositeFastFieldSerializer` is in charge of serializing
+/// fastfields on disk.
+///
+/// Fast fields have different encodings like bit-packing.
+///
+/// `FastFieldWriter`s are in charge of pushing the data to
+/// the serializer.
+/// The serializer expects to receive the following calls.
+///
+/// * `new_u64_fast_field(...)`
+/// * `add_val(...)`
+/// * `add_val(...)`
+/// * `add_val(...)`
+/// * ...
+/// * `close_field()`
+/// * `new_u64_fast_field(...)`
+/// * `add_val(...)`
+/// * ...
+/// * `close_field()`
+/// * `close()`
+pub struct CompositeFastFieldSerializer {
+    composite_write: CompositeWrite<WritePtr>,
+}
+
+impl CompositeFastFieldSerializer {
+    /// Constructor
+    pub fn from_write(write: WritePtr) -> io::Result<CompositeFastFieldSerializer> {
+        // just making room for the pointer to header.
+        let composite_write = CompositeWrite::wrap(write);
+        Ok(CompositeFastFieldSerializer { composite_write })
+    }
+
+    /// Serialize data into a new u64 fast field. The compression will be detected automatically.
```

Comment:
s/the compression will be detected automaticlly/The best compression codec will be chosen automatically.